src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,115 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { createComment } from '../api/posts';
+
+type Props = {
+  selectedPost: Post | null,
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>,
+  setCommentErrorMessage: (value: boolean) => void,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments, selectedPost, setCommentErrorMessage,
+}) => {
+  const [isFormLoading, setIsFormLoading] = useState(false);
+  const [commentName, setCommentName] = useState('');
+  const [commentEmail, setCommentEmail] = useState('');
+  const [commentText, setCommentText] = useState('');
```

Comment:
```suggestion
  const [newComment, setNewComment] = useState({
    name: '',
    email: '',
    text: '',
  });
```

suggest to make one state for all inputs and change proper field with event.target.name in one change handler instead three (key in state object should be the same as name attribute in input)
somethign like that
```
    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const { value, name } = event.target;
    setNewComment(prevNewComment => ({
        ...prevNewComment,
       [name]: value,
    }) );
  };
```